### PR TITLE
chore: upgrade onboard

### LINKED ISSUE #2017: Migrate to WalletConnect v2
## What is the feature about
On the 28th of June WalletConnect v1 will be shutdown.
We need to migrate `web3-onboard` to use WalletConnect v2.

## The list of requirements
Follow the [migration guide](https://docs.walletconnect.com/2.0/advanced/migration-from-v1.x/dapps#web3-onboard)

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Partially resolves #2017

## How this PR fixes it

This upgrades `@web3-onboard` and all associated modules to the latest version.

## How to test it

This likely requires a full regression - connect a wallet and expect the UI to work as before, especially transacting and switching between wallets.

## Checklist
* [x] I've tested the branch on mobile 📱
* [x] I've documented how it affects the analytics (if at all) 📊
* [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,14 +52,14 @@
...
-    "@web3-onboard/ledger": "^2.3.2",
-    "@web3-onboard/tallyho": "^2.0.1",
-    "@web3-onboard/trezor": "^2.3.2",
-    "@web3-onboard/walletconnect": "^2.1.3",
+    "@web3-onboard/coinbase": "^2.2.4",
+    "@web3-onboard/core": "2.19.2",
+    "@web3-onboard/injected-wallets": "^2.9.0",
+    "@web3-onboard/keystone": "^2.3.7",
+    "@web3-onboard/ledger": "^2.4.6",
+    "@web3-onboard/taho": "^2.0.4",
```
Comment (iamacook) [inline package.json:60]: `@web3-onboard/tallyho` was deprecated in favour of this.

File: jest.setup.js [hunk 2]
```diff
@@ -12,15 +12,7 @@ jest.mock('@web3-onboard/keystone/dist/index', () => jest.fn())
...
 jest.mock('@web3-onboard/walletconnect', () => jest.fn())
-jest.mock('@web3-onboard/tallyho', () => jest.fn())
-
-jest.mock('@web3-onboard/injected-wallets/dist/icons/metamask', () => '')
-jest.mock('@web3-onboard/coinbase/dist/icon', () => '')
-jest.mock('@web3-onboard/keystone/dist/icon', () => '')
-jest.mock('@web3-onboard/walletconnect/dist/icon', () => '')
-jest.mock('@web3-onboard/trezor/dist/icon', () => '')
-jest.mock('@web3-onboard/ledger/dist/icon', () => '')
-jest.mock('@web3-onboard/tallyho/dist/icon', () => '')
```
Comment (iamacook) [inline jest.setup.js:23]: We no longer load the icons from the modules.

**Comment by katspaugh** [comment]
Nice Chrome screenshot.

File: src/services/onboard.ts [hunk 3]
```diff
@@ -41,12 +40,18 @@ export const createOnboard = (chainConfigs: ChainInfo[], rpcConfig: EnvState['rp
...
     },
 
+    notify: {
+      enabled: false,
+    },
+
     appMetadata: {
       name: 'Safe{Wallet}',
-      icon: '/images/safe-logo-green.png',
+      icon: '<svg><image href="/images/safe-logo-green.png" width="56" height="56" /></svg>',
```
Comment (iamacook) [inline src/services/onboard.ts:49]: The logo does not render correctly in Chrome anymore unless wrapped in an SVG (as clarified [in the docs](https://onboard.blocknative.com/docs/modules/core#appmetadata)).

**Review by katspaugh — APPROVED** [review]

File: src/services/onboard.ts [hunk 4]
```diff
@@ -41,12 +40,18 @@ export const createOnboard = (chainConfigs: ChainInfo[], rpcConfig: EnvState['rp
...
       description: 'Please select a wallet to connect to Safe{Wallet}',
       recommendedInjectedWallets: getRecommendedInjectedWallets(),
     },
+
+    // TODO: Investigate using `autoConnectLastWallet` instead of our `lastWalletStorage`
```
Comment (katspaugh) [inline src/services/onboard.ts:54]: ❗ 👍

Comment (iamacook) [reply src/services/onboard.ts:54]: I've created [an issue](https://github.com/safe-global/safe-wallet-web/issues/2042) for this to keep track of it better.

**Comment by usame-algan** [comment]
Works well from my tests so far 👍 

Only some minor visual issues 

Before:
Screenshot 2023-05-26 at 12 23 23

After:
Screenshot 2023-05-26 at 12 22 50

**Comment by francovenica** [comment]
Does the Tally Ho wallet changed names? In prod is named Tally Ho Wallet, here is just Taho
image

There is a safe logo still broken when you choose a wallet to connect to:
image

The green icons showcasing the connected wallets are slightly bigger, making it touch the "t" in wallet connect
image

**Comment by katspaugh** [comment]
Taho rebranded: https://taho.xyz

**Comment by katspaugh** [comment]
We could decrease the font-size or letter-spacing but I think it looks fine.

**Comment by francovenica** [comment]
Trying to connect with the ledger causes an error. I tried to connect in current prod build v.1.12 and it works fine there.
I have a ledger nano S (I don't think the version matters, but still). 
If you connect to another wallet and try to connect to again to ledger doesn't show the error in the console, but shows an error in the modal (2nd snapshot)

image
image

File: src/hooks/wallets/useOnboard.ts [hunk 5]
```diff
@@ -145,31 +145,8 @@ export const connectWallet = async (
...
-      // Dismiss the modal
-      ;(onboardModal?.querySelector('.background') as HTMLElement)?.click()
-      tries = maxTries
-    }
-
-    tries += 1
-    if (tries >= maxTries) clearInterval(timer)
-  }, 100)
-}
-
```
Comment (iamacook) [inline src/hooks/wallets/useOnboard.ts:169]: This is [no longer required](https://github.com/blocknative/web3-onboard/issues/888#issuecomment-1487371407).

**Comment by iamacook** [comment]
> Trying to connect with the ledger causes an error.

Ledger compatibility is now fixed. All works fine with my Ledger Nano X on the latest firmware version.

**Comment by francovenica** [comment]
Tried with my ledger nano S and it works now.

Note that now is required to connect with ledger live, and the options you get for networks are more limited. In my case I can only pick Eth, Poly and BNB. So if people have safes  with the only owner being a ledger in GNO or OP or Arb then we should let them know about this change
image

**Comment by iamacook** [comment]
> The options you get for networks are more limited. In my case I can only pick Eth, Poly and BNB. 

I am unable to find this in the app. Can you explain where it is?

Some networks have their own Ledger apps to install: Polygon, Avalanche and CELO. Others would need the user to connect via WC (which Ledger Live natively supports) or MM:

image

> we should let them know about this change

We can adjust the chain configurations accordingly.

**Pull request merged**